### DISCUSSION TITLE
pyscript Installation Problem

### DISCUSSION THREAD

**Discussion by Schevenhuette** [custom-components/pyscript#250] [Q&A]
Hi, I am quite new working with Home Assistant, installed on an Pi4.
I wanted to use a Python script that uses the **math** library. As far as I understand, this is only possible with the HACS extension **pyscript**. When trying to install I have the following problem:

I tried to do like described here [https://hacs-pyscript.readthedocs.io/en/stable/installation.html](url)
Option 1 didn't work cause **pyscript** could not be found
Option 2 seemd to work at first but so fardid not!
The directotry **config/custom_components** already existed. **hacs** is in there.
I made the folder **config/custom_components/pyscript** and unzipped **hass-custom-pyscript.zip** in there.

By the way, I skipped to install the **Jupyter kernel** cause it is optional and I do not have a clue what it is yet.

Configuration via **UI** did not work because I could not find **Pyscript Python scripting** under Configuration/Integration neither under HACS/Integrations.
I choose the alternative adding **pyscript:** to **<config>/configuration.yaml** with both parameters as told.

Next I created the folder **config/pyscript** and moved my script **dew_point.py** (uses **math**) from folder python_scripts in there. **Conviguration validation** was ok so I restarted HASS.

![image](https://user-images.githubusercontent.com/73844974/135740023-c559404b-49f0-4d6c-8347-902eaa623eb2.png)

Now these script can not be called under **Developer Tools Services** so far.
Located in folder python_scripts it can but produces following LOG entry

Logger: homeassistant.components.python_script.dew_point.py
Source: components/python_script/__init__.py:222
Integration: Python Scripts (documentation, issues)
First occurred: 6:41:23 AM (1 occurrences)
Last logged: 6:41:23 AM

Error executing script: __import__ not found
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/python_script/__init__.py", line 222, in execute
    exec(compiled.code, restricted_globals)
  File "dew_point.py", line 12, in <module>
ImportError: __import__ not found

Now I am wondering why I install pyscript in folder **config/custom_components/pyscript** and add **pyscript:** to my **<config>/configuration.yaml**. Anyway I tried to unzip **hass-custom-pyscript.zip** in **config/pyscript**, did not work.
I tried to add **custom_components/pyscript:** (exactly like this) to my **<config>/configuration.yaml** ending up that the **Conviguration validation** was not ok so I decided not to restart HASS.

Can anybody help me with that?
Thanks a lot already, Thomas

**Comment by stefanuytterhoeven** [comment]
Why not install HACS first? Since you're new, it's probably easier. Try to find out how to install HACS (Home assistant community store)
HACS is some kind of "store" of "integrations", "frontends", .... and.... it enables you to install PYSCRIPT.

**Reply by Schevenhuette** [reply]
Thanks for your answer but yes I installed HACS first. I did not know there is a way to install **Pyscript** without install HACS first.
Anyway, by now I tried **Appdaemon** and it worked out better so far.

**Comment by dlashua** [comment]
Appdaemon is a great system as well. Two very different approaches. Appdaemon is class based, and lives outside of Home Assistant, therefore, it can only do things the Home Assistant Websocket allows. Pyscript is more decorator based, and it lives inside of Home Assistant with full access to HASS internals. In most cases, however, you can do the same things with either one.

**Comment by mudkipdev** [comment]
Did you ever fix this? I am facing the same problem as you and I have taken the exact steps as you too.

**Comment by stefanuytterhoeven** [comment]
Just install hacs, then install pyscript through hacs.
Works without any problem

**Reply by politas** [reply]
Except no, it doesn't. I too have HACS installed but cannot find pyscripts. The only things that come up searching for "py" under Home Assistant Community Add-ons are AppDaemon and JupyterLab. Searching Integrations for "py" brings up a few options, including "Python Scripts" and "Remote Python Debugger", but not "pyscripts"

**Reply by davidbd7** [reply]
Did you fix this issue, I am having a similar one

**Reply by politas** [reply]
Yes, it turns out that "Home Assistant Community Add-ons" and "HACS" are entirely separate things.

**Comment by GaupeLnx** [comment]
I have (just a few weeks ago) installed pyscript (not pyscripts)  via HACS  and i must say it works just perfect.   
you dont need to install the jupyter part. 
But with jupyter, it does make it a lot easyer to write and debug pyscripts.
You then can do it just from your pc and run and debug it in visual studio code, and then when you got the script to your liking upload it to Home assistant.  
If you can't find it in HACS i would first look at if there is something wrong with the HACS installation.
